src/lib/slider/slider.ts

```diff
@@ -182,6 +182,7 @@ export class MdSlider implements AfterContentInit, ControlValueAccessor {
 
     this.isActive = true;
     this.isSliding = false;
+    this._sliderDimensions = this._renderer.getSliderDimensions();
```

Comment:
Hi @jelbourn, two things:

- to make the unit tests pass, `_sliderDimensions` needs to be updated in both `onClick` and `onSlideStart`, though it may be because the tests generate the click/slide events explicitly.
- initially I couldn't write a **failing** test, I believe it is due to the way the click/slide events are generated in the tests.

The click/slide events in the unit tests take a "percentage", grab the **up-to-date** dimensions of the element and calculate the event coordinates based on that, which is backwards to how the browser does it. This means the values are always "correct" even if you move the sliders' parent around. See: https://github.com/angular/material2/blob/master/src/lib/slider/slider.spec.ts#L688

Adjusting these methods to take (pre-calculated) coordinates of an event, tests fail/pass as expected. Am I missing something obvious though...?